src/main/java/Duke.java

```diff
@@ -1,10 +1,50 @@
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
+        ui.welcomeMessage();
+        tasks.showCurrentTasks();
+        ui.showLineBreak();
+        boolean isFinished = false;
+        while (!(isFinished)) {
+            try {
+            String inputFromUser = ui.handleInput();
+               Command c = Parser.parse(inputFromUser);
+            if (c == null) {
+                throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-( ☹ OOPS!!!");
+            }
+               tasks.runCommand(c);
+            if (c.getCommand().equals("bye")) {
```

Comment:
Instead of saving what type of command is it as a string, maybe you could use 'instanceof' to know that is a 'ByeCommand'?